dbms/src/DataStreams/ProhibitColumnsBlockOutputStream.cpp

```diff
@@ -13,9 +13,12 @@ namespace ErrorCodes
 
 void ProhibitColumnsBlockOutputStream::write(const Block & block)
 {
-    for (const auto & column : columns)
-        if (block.has(column.name))
-            throw Exception{"Cannot insert column " + column.name, ErrorCodes::ILLEGAL_COLUMN};
+    if (allow_materialized)
```

Comment:
With `allow_materialized`, `ProhibitColumnsBlockOutputStream` effectively becomes no-op.
It's better to not use `ProhibitColumnsBlockOutputStream` in INSERT pipeline in this case at all, look at `InterpreterInsertQuery.cpp`:

```
out = std::make_shared<ProhibitColumnsBlockOutputStream>(out, table->materialized_columns);
```

You just have to add a condition above this line.
And `ProhibitColumnsBlockOutputStream` will be unmodified.

Reply:
And you have to do the same at `InterpreterCreateQuery.h` for the case of `CREATE ... SELECT`.